cmd/gardener-scheduler/app/gardener_scheduler.go

```diff
@@ -0,0 +1,369 @@
+	return nil
+}
+
+func (o *Options) applyDefaults(in *config.SchedulerConfiguration) (*config.SchedulerConfiguration, error) {
+	external, err := scheme.Scheme.ConvertToVersion(in, schedulerconfigv1alpha1.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	scheme.Scheme.Default(external)
+
+	internal, err := scheme.Scheme.ConvertToVersion(external, config.SchemeGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	out := internal.(*config.SchedulerConfiguration)
+
+	return out, nil
+}
+
+func (o *Options) run(ctx context.Context, cancel context.CancelFunc) error {
+	if len(o.ConfigFile) > 0 {
+		c, err := o.loadConfigFromFile(o.ConfigFile)
+		if err != nil {
+			return err
+		}
+		o.config = c
+	}
+
+	gardener, err := NewGardenerScheduler(o.config)
+	if err != nil {
+		return err
+	}
+
+	return gardener.Run(ctx, cancel)
+}
+
+// NewCommandStartGardenerScheduler creates a *cobra.Command object with default parameters
+func NewCommandStartGardenerScheduler(ctx context.Context, cancel context.CancelFunc) *cobra.Command {
+	opts, err := NewOptions()
+	if err != nil {
+		panic(err)
+	}
+
+	cmd := &cobra.Command{
+		Use:   "gardener-scheduler",
+		Short: "Launch the Gardener scheduler",
+		Long:  `The Gardener scheduler is a controller that tries to find the best matching seed cluster for a shoot. The scheduler takes the cloud provider and the distance between the seed (hosting the control plane) and the shoot cluster region into account.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			utilruntime.Must(opts.configFileSpecified())
+			utilruntime.Must(opts.validate(args))
+			utilruntime.Must(opts.run(ctx, cancel))
+		},
+	}
+
+	opts.config, err = opts.applyDefaults(opts.config)
+	if err != nil {
+		panic(err)
+	}
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+// GardenerScheduler represents all the parameters required to start the
+// Gardener scheduler.
+type GardenerScheduler struct {
+	Config                 *config.SchedulerConfiguration
+	Identity               *gardenv1beta1.Gardener
+	GardenerNamespace      string
+	K8sGardenClient        kubernetes.Interface
+	K8sGardenInformers     gardeninformers.SharedInformerFactory
+	K8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
+	KubeInformerFactory    informers.SharedInformerFactory
+	Logger                 *logrus.Logger
+	Recorder               record.EventRecorder
+	LeaderElection         *leaderelection.LeaderElectionConfig
+}
+
+// NewGardenerScheduler is the main entry point of instantiating a new Gardener Scheduler.
+func NewGardenerScheduler(cfg *config.SchedulerConfiguration) (*GardenerScheduler, error) {
+	if cfg == nil {
+		return nil, errors.New("config is required")
+	}
+
+	// validate the configuration
+	if err := validation.ValidateConfiguration(cfg); err != nil {
+		return nil, err
+	}
+
+	// Initialize logger
+	logger := logger.NewLogger(cfg.LogLevel)
+	logger.Info("Starting Gardener scheduler ...")
+
+	if flag := flag.Lookup("v"); flag != nil {
```

Comment:
If you want to be able to incorporate kubernetes logging options, you can do so by adding `flag.CommandLine` to your `cobra.Command` and doing an `_ import` to where the kubernetes logging flags are being defined. This way you don't have to do manual parsing here.

Reply:
Thanks! After looking over it again, I decided to remove it because currently k8s.io/klog is not used in the scheduler.